knock_knock/knock_knock/utils.py

```diff
@@ -70,6 +70,23 @@ def minute_docket_scheduler():
 				if whatsapp_number:
 					send_whatsapp_msg(whatsapp_number, docket_doc.description+ '\nReference: ' + docket_url, docket_doc.doctype, docket_doc.name)
 
+@frappe.whitelist()
+def monthly_docket_scheduler():
+	current_date_time = get_datetime(now())
+	dockets = frappe.db.get_all('Docket', filters = { 'status': 'Open', 'repeat_on': 'Monthly'})
+	if dockets:
+		for docket in dockets:
+			docket_doc = frappe.get_doc('Docket', docket.name)
+			whatsapp_number = frappe.db.get_value('User', docket_doc.owner, 'user_whatsapp_number')
+			docket_url = get_url_to_form(docket_doc.doctype, docket_doc.name)
+			docket_due_message_mail = 'Docket '+ docket_doc.name + ' for '+ docket_doc.subject + ' must be completed before'+ str(docket_doc.due_date)
+			docket_due_message_wtsp = 'Docket '+ docket_doc.name + ' for '+ docket_doc.subject + ' must be completed before *'+ str(docket_doc.due_date) + '*.\nReference : ' + docket_url
+			create_notification_log(docket_doc.subject, docket_doc.owner, docket_doc.description, docket_doc.doctype, docket_doc.name)
+			if whatsapp_number:
+				send_whatsapp_msg(whatsapp_number, docket_doc.description+ '\nReference: ' + docket_url, docket_doc.doctype, docket_doc.name)
+
+
+
```

Comment:
@Aswani-m Make the function generic and apply conditions for all repeat on conditions

Reply:
Done. Please check